examples/api-routes-apollo-server-and-client/apollo/resolvers.js

```diff
@@ -1,7 +1,7 @@
 export const resolvers = {
   Query: {
     viewer (_parent, _args, _context, _info) {
-      return { name: 'John Smith', id: 1 }
+      return { id: 1, name: 'John Smith', status: 'cached' }
```

Comment:
@HaNdTriX the status is not on the server, the whole point of having the status was to make sure apollo client was setup correct for both client and server.

Reply:
First of all Thanks for your feedback! 🙏

Afaik it was rendered on the server in the example before.
So the `@client` directive didn't had any effect.

**SSR Response:** (example before this PR)

![Bildschirmfoto 2019-09-11 um 01 52 09](https://user-images.githubusercontent.com/1265681/64658228-085b4600-d437-11e9-9a0b-d68d842fca3a.png)

In addition to that you had to trigger the query manually in: https://github.com/zeit/next.js/blob/canary/examples/api-routes-apollo-server-and-client/pages/index.js#L36-L38. Normally `getDataFromTree` should run all queries used in the react tree. 

Can you further explain your the propose of your technique. Maybe I have a knowledge gap here 😬.

Reply:
In the [example before](https://github.com/zeit/next.js/blob/canary/examples/api-routes-apollo-server-and-client) you can actually remove the following code without noticing any change in behaviour:

https://github.com/zeit/next.js/blob/canary/examples/api-routes-apollo-server-and-client/lib/with-apollo-client.js#L23-L41

Reply:
@HaNdTriX this is the relevant code that is now broken.

https://github.com/zeit/next.js/blob/5496e0f8b380a05081972e60c07ada8be605c9e5/examples/api-routes-apollo-server-and-client/lib/init-apollo.js#L27-L38

it was covering the case when people want to use local state: https://www.apollographql.com/docs/react/essentials/local-state/

Reply:
thats why there was an `@client` directive

Reply:
Please note that the `@client` directive targets the `ApolloClient` wich is present on the server **and** on the client (in ssr). This is why `getDataFromTree` was not working in your example.

Reply:
@HaNdTriX which is probably an issue we should bring up to the apollo team, because with this setup, it would make sense to run here since it's really just a value in the cache.

Reply:
@lfades any suggestions here?

Reply:
I am currently not sure how to properly solve this issue, since the `ApolloClient` needs to know the entire schema in order to resolve all queries (server & browser-client) ones. Bringing the issue of local state and ssr to the apollo teams seems as a good idea to me.

May I resolve this conversation so we can move on with this PR?

Reply:
IMO the local state should work with `getDataFromTree` and it shouldn't be an issue, otherwise that's an Apollo issue. But if it's currently breaking the example then I don't mind not having it, I don't really know how many people is using the local state in their applications though. I tried using it once and error handling was a nightmare.

Reply:
looks like there might be a bug in `getDataFromTree` https://github.com/apollographql/react-apollo/issues/3338

I also opened a separate issue for client state issue. https://github.com/apollographql/react-apollo/issues/3486

Reply:
@lfades would love to hear more about your experience with apollo local state, I'm currently using it in a project at work, and it *mostly* works although its kind of broken in SSR and trying to get the Apollo team to look at issues on GitHub seems near impossible :|

Reply:
Hey @lifeiscontent 

I believe localState actually works fine with Next.js.
It's important to understand, that the `@client` directive targets the `ApolloClient` and so a react app as a whole. This makes totally sence, since you can't render an app on the server if some parts of the schema or of your data is missing.

Therefor the localState (`resolvers` & `typeDefs`) must be present on the server as well as on the client.

I've created an example for your that works fine. Without additional queries. Please check out the following commit (branch): https://github.com/HaNdTriX/next.js/commit/1edae98cc8822cdfbdb05197851a181585753f81

If you run the example and press the `Show More` Button to see client state in action.

I didn't dig deeper into your example with the `SchemaLink`. Is it possible that this `SchemaLink` interferes the `clientSchema` so it breaks `getDataFromTree`?

Reply:
@HaNdTriX can you put your commit on a branch? I'm unsure how to checkout a remote commit without a branch.

Reply:
https://github.com/HaNdTriX/next.js/tree/examples/apollo-localState

Branch: examples/apollo-localState